src/engraving/rw/compat/compatutils.cpp

```diff
@@ -339,6 +340,52 @@ void CompatUtils::reconstructTypeOfCustomDynamics(MasterScore* score)
     }
 }
 
+void CompatUtils::splitArticulations(MasterScore* score)
+{
+    std::vector<Articulation*> toRemove;
+    for (Measure* meas = score->firstMeasure(); meas; meas = meas->nextMeasure()) {
+        for (Segment& seg : meas->segments()) {
+            if (!seg.isChordRestType()) {
+                continue;
+            }
+            for (EngravingItem* item : seg.elist()) {
+                if (!item || !item->isChord()) {
+                    continue;
+                }
+                Chord* chord = toChord(item);
+                for (Articulation* a : chord->articulations()) {
+                    std::set<SymId> ids = mu::engraving::splitArticulations({ a->symId() });
+                    if (ids.size() <= 1) {
+                        continue;
+                    }
+                    toRemove.push_back(a);
+                }
+            }
+        }
+    }
+    for (Articulation* combinedArtic : toRemove) {
+        auto components = mu::engraving::splitArticulations({ combinedArtic->symId() });
+        Chord* parentChord = toChord(combinedArtic->parentItem());
+        for (SymId id : components) {
+            Articulation* newArtic = Factory::createArticulation(score->dummy()->chord());
+            newArtic->setSymId(id);
+            if (parentChord->hasArticulation(newArtic)) {
+                delete newArtic;
+                continue;
+            }
+            newArtic->setParent(parentChord);
+            newArtic->setTrack(combinedArtic->track());
+            newArtic->setPos(combinedArtic->pos());
+            newArtic->setDirection(combinedArtic->direction());
+            newArtic->setLinks(combinedArtic->links());
```

Comment:
So the distinct components (which should be considered independent elements, right?) will get the same link. That seems not expected to me, is it?

Reply:
I was not entirely certain how the links work. I was worried about excerpts getting messed up by this, but I wasn't sure how to deal with it. Do you have any advice as to how to do this without messing up excerpts?

Reply:
Suppose you get an old score with the following situation:
- on staff 1, there is a Chord with a compound articulation, say, an accent-staccato
- staff 2 is linked to staff 1
- there is also a part score that is linked to staff 1

These three instances of the compound articulation are all linked together. So, there is a `Link` containing 3 items.

Now, the compound articulation needs to be split into an accent and a staccato. Each of the two new articulations should become linked to its _respective_ counterparts. So, in the end you should have two `Links`:
- Link 1:
    - the accent in staff 1 of the score
    - the accent on the linked staff 2
    - the accent in the linked part score
- Link 2: 
    - the staccato in staff 1 of the score
    - the staccato on the linked staff 2
    - the staccato in the linked part score

So a link should _basically_ only ever contain representatives of exactly the same musical element.

---

_However_, for _some_ of the migration logic in `CompatUtils`, where an element of one type has to be replaced one element of another type, @mike-spa had a slight hack to make this easier: suppose you get the following situation:
- Link 1:
    - representative 1 of the old item
    - representative 2 of the old item

Then, we add the new items to this link, so we are temporarily in an inconsistent state:
- Link 1:
    - representative 1 of the old item
    - representative 2 of the old item
    - representative 1 of the replacement item
    - representative 2 of the replacement item

But then we delete the old items and we're happy again:
- Link 1:
    - representative 1 of the replacement item
    - representative 2 of the replacement item

This idea works great if you need to replace one element with one other. But if you need to replace one element with two, it's not gonna work, because the two replacement elements need to be in distinct links (because they are simply distinct elements).

Reply:
That is such a help, I really appreciate you writing all of that and I now understand completely. I'll fix that next week and we should be good to go. As it stands there are a few more points that this PR needs to cover before it's 'ready' anyway. Thanks again!